### DISCUSSION TITLE
How to Start your Opensource contributions on GitHub as a Beginner

### DISCUSSION THREAD

**Discussion by AngelofVerdant** [community/community#42141] [New to GitHub]
I am writing this as a beginner myself to other beginners out there. Contributing to open-source projects on GitHub is one of those things that makes for a  good resume. So how do you do it as a beginner especially if you're still not very good with code? Here are a few pointers:

1. Find Opensource projects that have incomplete readme files, then do a detailed description of what the project is all about, well outlined, more of like a tutorial to guide first-timers. Go the extra mile and include screenshots and sketches  and trust me, you will succeed and get your Pull Request Merged

2. The Second thing is a bit simple but also works, go through project descriptions and try to find incoherent statements and typos and fix them. It's simple but it does get you into that contribution list.

4. The third and final one that also works like a charm is to choose a project you love and see if it is available in your language, if it is not, then that is your chance. Add Support for the language of your choice. Usually done by editing the strings.xml file that is provided. 

There might be many other ways to contribute but so far these are the three ways I have used to get my name on the contribution list of open-source projects. I do hope this finds somebody and helps them.

**Comment by Luis-Angel-Santos** [comment]
It seems to me a good way if you don't have any knowledge in programming, but it would also include making small modifications in small programs or small parts of the code that don't affect seriously. This way we will go from less to more.

**Comment by queenofcorgis** [comment]
Hey there @AngelofVerdant - congrats on starting your GItHub journey 🎉 

Recommend checking out the latest [New to GitHub Check-in ](https://github.com/community/community/discussions/42302)- we spotlight a collection of projects great for beginner contributors.

**Reply by AngelofVerdant** [reply]
Thanks , and sure thing

**Comment by codeByunique** [comment]
Loved this post! 🙌  
I'm a backend dev learning GitHub contributions. What’s the best way to find good-first-issues that actually get reviewed?

**Comment by Shantanugupta43** [comment]
Hi if anyone's interested in beginner friendly open source Chrome extension feel free to contribute 

https://github.com/Shantanugupta43/SuggestPilot

Tech stack HTML, CSS, JavaScript

**Comment by SaiVatsal** [comment]
Improve documentation: add instructions, tutorials, screenshots, diagrams, etc. to the README file.

Fix typos & grammar: correct incoherent statements or small mistakes in the project description.

Add translations: contribute language support by editing the file "strings.xml."

Other easy contributions: test the project, report bugs, organize issues, add examples, etc. 

Docs → Improve the README file, add tutorials/screenshots.

Typos → Fix grammatical mistakes or incoherent statements.

Translations → Add language support, edit "strings.xml."

Extras → Test the project, report bugs, organize issues, add examples, etc.

**Comment by AhndreWalters** [comment]
Hey there!

Thank you for this. As someone also trying to break into open source, this is exactly what I needed. You've perfectly tackled the biggest fear most beginners have: "I'm not good enough at coding to contribute."

I'd add one more tip: look for issues labeled good first issue or help wanted on repositories. These are tagged specifically for beginners and are a great starting point.

Your three tips are honestly underrated and super effective. Most maintainers love when someone improves their documentation. Thanks for sharing this!
